### PR TITLE
Script for defect 2457

### PR DESCRIPTION
**Pull request by TMyza**
ATF Test Scripts to check [#2457](https://github.com/smartdevicelink/sdl_core/issues/2457)

Summary
A set of test scripts to verify that issue is fixed

### CODE REVIEW

File: test_scripts/Defects/5_0/2457_Needed_fixes_sending_of_UpdateDeviceList_notification_on_device_connect.lua [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+    commonFunctions:waitForSDLStart(test)
+    :Do(function()
+        test:initHMI()
+        :Do(function()
+            utils.cprint(35, "HMI initialized")
+            test:initHMI_onReady(pHMIParams)
+            :Do(function()
+                utils.cprint(35, "HMI is ready")
+                common.getHMIConnection():ExpectNotification("BasicCommunication.UpdateDeviceList")
+                common.getHMIConnection():ExpectNotification("OnDeviceAdded")
```
Comment (GetmanetsIrina) [inline test_scripts/Defects/5_0/2457_Needed_fixes_sending_of_UpdateDeviceList_notification_on_device_connect.lua]: please clarify requirements regarding the `OnDeviceAdded`, maybe is not actual for now or for open SDL

File: test_scripts/Defects/5_0/2457_Needed_fixes_sending_of_UpdateDeviceList_notification_on_device_connect.lua [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+    return common.getHMIConnection():ExpectEvent(event, "Start event")
+end
+
+--[[ Scenario ]]
+runner.Title("Preconditions")
+runner.Step("Clean environment", common.preconditions)
+
+-- [[ Test ]]
+runner.Title("Test")
+runner.Step("Start SDL, HMI, connect Mobile, start Session", start)
```
Comment (GetmanetsIrina) [inline test_scripts/Defects/5_0/2457_Needed_fixes_sending_of_UpdateDeviceList_notification_on_device_connect.lua]: Session is not started in `start` function

File: test_scripts/Defects/5_0/2457_Needed_fixes_sending_of_UpdateDeviceList_notification_on_device_connect.lua [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+
+-- [[ Postconditions ]]
+runner.Title("Postconditions")
+runner.Step("Stop SDL", common.postconditions)
```
Comment (GetmanetsIrina) [inline test_scripts/Defects/5_0/2457_Needed_fixes_sending_of_UpdateDeviceList_notification_on_device_connect.lua]: Please add new line in the end.

File: test_scripts/Defects/5_0/2457_Needed_fixes_sending_of_UpdateDeviceList_notification_on_device_connect.lua [hunk 4]
```diff
@@ -0,0 +1,72 @@
+---------------------------------------------------------------------------------------------------
+-- User story: https://github.com/smartdevicelink/sdl_core/issues/2381
+--
+-- Precondition:
+-- SDL Core and HMI are started. App is registered, HMI level = FULL
+-- Description:
+-- Steps to reproduce:
```
[the remark this answers is not in the document]
Comment (TMyza) [reply test_scripts/Defects/5_0/2457_Needed_fixes_sending_of_UpdateDeviceList_notification_on_device_connect.lua]: Fixed in: https://github.com/smartdevicelink/sdl_atf_test_scripts/pull/2036/commits/4abb48c35f7827ec6610fc1570f9bac0c65ce52c

**Comment by GetmanetsIrina** [comment]
approved.

**Pull request merged**